### PR TITLE
add browser based freighter-api and update docs

### PR DESCRIPTION
**Pull request by piyalbasu**
creating an API for non-node applications. This will be published on cdnjs, similar to stellar-sdk

### CODE REVIEW

File: @stellar/freighter-api/src/index.ts [hunk 1]
```diff
@@ -5,3 +5,10 @@ import { getNetwork } from "./getNetwork";
 
 export { getPublicKey, signTransaction, isConnected, getNetwork };
 export default { getPublicKey, signTransaction, isConnected, getNetwork };
+
+window.freighterApi = {
+  getPublicKey,
+  signTransaction,
+  isConnected,
+  getNetwork,
+};
```
Comment (piyalbasu) [inline @stellar/freighter-api/src/index.ts:14]: this is the big change to the API. 

cdnjs will package this file and host on their server, so we will expose both the es modules for npm users as well as hanging the methods off of `window` for other apps

**Comment by shanzzam** [comment]
This is a great addition!

File: docs/docs/guide/gettingStarted.md [hunk 2]
```diff
@@ -16,8 +16,14 @@ You'll want a local version of the extension to test with.
...
+
 - Install the module using npm: `npm install @stellar/freighter-api`
 
 or
 
 - Install the module using yarn: `yarn add @stellar/freighter-api`
+
+#### For browser-based applications
+
+- Install the packaged library via script tag using cdnjs: `<script src="https://cdnjs.cloudflare.com/ajax/libs/stellar-freighter-api/{version}/index.min.js"></script> `
```
Comment (leighmcculloch) [inline docs/docs/guide/gettingStarted.md:29]: Looks like there is a stray space on this. Also, fyi you can make this a styled code block and still keep it in a list form:
````suggestion
- Install the packaged library via script tag using cdnjs:
   ```html
   <script src="https://cdnjs.cloudflare.com/ajax/libs/stellar-freighter-api/{version}/index.min.js"></script>
   ```
````

File: docs/docs/guide/usingFreighterBrowser.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+We now have an extension installed on our machine and a library to interact with it. This library will provide you methods to send and receive data from a user's extension in your website or application.
+
+### Importing
+
+First import the library in the `<head>` tag of your page.
+
+```html
+<head>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/stellar-freighter-api/{version}/index.min.js"></script>
```
Comment (quietbits) [inline docs/docs/guide/usingFreighterBrowser.md:14]: Do users need to enter `version` manually? If so, maybe it's worth adding a note here about it.

File: docs/docs/guide/usingFreighterBrowser.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+### signTransaction
+
+#### `signTransaction(xdr: string, network:? string) -> <Promise<string>>`
+
+This function accepts a transaction XDR string as the first parameter, which it will decode, sign as the user, and then return the signed transaction to your application.
+
+The user will need to provide their password if the extension does not currently have their private key. Once the user has provided their password, the extension will have access to the user private key for 5 minutes. The user must then review the transaction details and accept within those 5 minutes for the transaction to be signed.
+
+_NOTE:_ Then user must provide a valid transaction XDR string for the extension to properly sign.
```
Comment (marcelosalloum) [inline docs/docs/guide/usingFreighterBrowser.md:106]: nit:
```suggestion
_NOTE:_ The user must provide a valid transaction XDR string for the extension to properly sign.
```

File: docs/docs/guide/usingFreighterBrowser.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+if (window.freighterApi.isConnected()) {
+  alert("User has Freighter!");
+}
+
+const retrievePublicKey = async () => {
+  let publicKey = "";
+  let error = "";
+
+  try {
+    publicKey = await getPublicKey();
```
Comment (quietbits) [inline docs/docs/guide/usingFreighterBrowser.md:52]: Do we need to add `window.freighterApi` to `getPublicKey()`?

File: @stellar/freighter-api/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@stellar/freighter-api",
-  "version": "1.0.1",
+  "version": "1.2.0",
```
Comment (marcelosalloum) [inline @stellar/freighter-api/package.json:3]: Why are we going to 1.2.0? Shouldn't it be 1.1.0?

**Review by quietbits — APPROVED** [review]

**Review by marcelosalloum — APPROVED** [review]
LGTM! Great improvement 🙌 .

I left a nit and one question below ⬇️

**Review by leighmcculloch — APPROVED** [review]
Woohoo!

File: @stellar/freighter-api/package.json (hunk 6 above)
Comment (leighmcculloch) [reply @stellar/freighter-api/package.json:3]: This change looks like it is adding new capabilities, not fixing existing capabilities, so I'd vote it is a minor version bump, i.e. 1.2.0.

Comment (piyalbasu) [reply @stellar/freighter-api/package.json:3]: Bah, you're right. Should be `1.1.0`

**Pull request merged**